src/rp2_common/pico_audio_i2s/audio_i2s.c

```diff
@@ -52,10 +52,23 @@ const audio_format_t *audio_i2s_setup(const audio_format_t *intended_audio_forma
     gpio_set_function(config->clock_pin_base, func);
     gpio_set_function(config->clock_pin_base + 1, func);
 
+#if PICO_PIO_USE_GPIO_BASE
+    if(config->data_pin >= 32 || config->clock_pin_base + 1 >= 32) {
+        pio_set_gpio_base(audio_pio, 16);
```

Comment:
Should this also assert that `(config->data_pin >= 16 && config->clock_pin_base >= 16)` ? (or can we assume that people will always configure their pins in a compatible manner?)

Reply:
I added the requested assertion.